qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1232,26 +1238,24 @@ private ResolvedJavaTypeInfo validateJavaTypePart(Part part, Section ownerSectio
 			return null;
 		}
 
-		return validateIterable(part, ownerSection, resolvedJavaType, javaTypeToResolve, diagnostics);
+		return resolvedJavaType;
 	}
 
 	private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 			ResolvedJavaTypeInfo resolvedJavaType, String javaTypeToResolve, List<Diagnostic> diagnostics) {
-		if (part != null && part.isLast() && ownerSection != null && ownerSection.isIterable()) {
-			// The expression is declared inside an iterable section like #for, #each.
-			// Ex: {#for item in items}
-			if (!resolvedJavaType.isIterable() && !resolvedJavaType.isInteger()) {
-				// The Java class is not an iterable class like
-				// - java.util.List
-				// - object array
-				// - integer
-				String expression = part.getParent().getContent();
-				Range range = QutePositionUtility.createRange(part);
-				Diagnostic diagnostic = createDiagnostic(range, DiagnosticSeverity.Error, QuteErrorCode.IterationError,
-						expression, javaTypeToResolve);
-				diagnostics.add(diagnostic);
-				return null;
-			}
+		// The expression is declared inside an iterable section like #for, #each.
+		// Ex: {#for item in items}
+		if (!resolvedJavaType.isIterable() && !resolvedJavaType.isInteger()) {
+			// The Java class is not an iterable class like
+			// - java.util.List
+			// - object array
+			// - integer
+			String expression = part.getParent().getContent();

```

Comment:
you previously had a null-check on part. is it not necessary anymore?

Reply:
The iterable check is done now only when last part of a #for or #each section is scanned. In this case we are sure that part is never null and the section is an iterable section.